core/opendaq/component/include/opendaq/tags_impl.h

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <opendaq/tags_ptr.h>
+#include <opendaq/tags_private_ptr.h>
+#include <opendaq/component_ptr.h>
+#include <coretypes/intfs.h>
+#include <coretypes/validation.h>
+#include <coreobjects/eval_value_factory.h>
+#include <coreobjects/core_event_args_impl.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+class TagsImpl : public ImplementationOf<ITags, ITagsPrivate, ISerializable>
+{
+public:
+    explicit TagsImpl();
+    explicit TagsImpl(const ProcedurePtr& coreEventCallback);
+
+    // ITags
+
+    ErrCode INTERFACE_FUNC getList(IList** value) override;
+    ErrCode INTERFACE_FUNC add(IString* name) override;
+    ErrCode INTERFACE_FUNC remove(IString* name) override;
+    ErrCode INTERFACE_FUNC contains(IString* name, Bool* value) override;
+    ErrCode INTERFACE_FUNC query(IString* query, Bool* value) override;
+    
+    // IBaseObject
+
+    ErrCode INTERFACE_FUNC equals(IBaseObject* other, Bool* equals) const override;
```

Comment:
If we have equals method implemented, then we should also think about implementing the getHashCode()